src/sail_lean_backend/pretty_print_lean.ml

```diff
@@ -165,6 +165,7 @@ let rec doc_typ ctx (Typ_aux (t, _) as typ) =
   | Typ_app (Id_aux (Id "range", _), [A_aux (A_nexp low, _); A_aux (A_nexp high, _)]) ->
       if provably_nneg ctx low then string "Nat" else string "Int"
   | Typ_var kid -> doc_kid ctx kid
+  | Typ_app (id, _) -> doc_id_ctor id
```

Comment:
`doc_id_ctor` is being used for both type constructors and datatype constructors (both here and in code already merged).  Is that intentional?

Reply:
I think it does the right thing for both until now, maybe it should just be called `doc_id`

Reply:
OK, will merge now.